### PR TITLE
[5.1] [Typechecker] Dont infer @objc for typealiases inside @objc protocol

### PR DESCRIPTION
**Pull request by theblixguy**
Cherry pick of #23346.

Dont infer @objc for typealiases inside @objc protocol. This stops the assertion from being hit inside Serialization.cpp.

Resolves [SR-10115](https://bugs.swift.org/browse/SR-10115).

### CODE REVIEW

**Comment by theblixguy** [comment]
cc @jrose-apple

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Review by jrose-apple — APPROVED** [review]

**Review by shahzadlone — APPROVED** [review]

**Comment by theblixguy** [comment]
Can this be merged now or does it need approval from Anna?

**Comment by jrose-apple** [comment]
Officially it needs approval from Ted or someone Ted's delegated to. I'll tag in some more potential approvers.

**Review by bob-wilson — APPROVED** [review]

**Pull request merged**
